README.md

```diff
@@ -8,15 +11,71 @@ If you want a module that specifically aims to be a complete and transparent int
 
 Initially this module was named Net::LDNS.
 
-IDN
-===
-If GNU libidn is installed when this module is compiled, it will be used to add a simple function that converts strings from Perl's internal encoding to IDNA domain name format. In order to convert strings from whatever encoding you have to Perl's internal format, use L<Encode>. If you need any kind of control or options, use L<Net::LibIDN>. The included function here is only meant to assist in the most basic case, although that should cover a lot of real-world use cases.
 
-Installation
-============
+## Features
+
+When installing from source, you can choose to enable or disable a number
+of optional features using command line options to the `perl Makefile.PL`
+commands.
+
+### IDN
+
+This feature is enabled by default.
+Disable with `--no-idn`.
+
+If the IDN feature is enabled, the GNU `libidn` library will be used to
+add a simple function that converts strings from Perl's internal encoding
+to IDNA domain name format.
+In order to convert strings from whatever encoding you have to Perl's
+internal format, use L<Encode>.
+If you need any kind of control or options, use L<Net::LibIDN>.
+The included function here is only meant to assist in the most basic case,
+although that should cover a lot of real-world use cases.
+
+> **Note:** The Zonemaster Engine test suite assumes this feature
+> is enabled.
+
+### Randomized capitalization
+
+This feature is disabled by default.
+Enable with `--randomize`.
+
+Enables compilation with randomzied capitalization.
```

Comment:
It might be that Zonemaster requires that the randomization is turned off. Do we have any reason to make optional instead of just have it turned off? If we have it selectable we should test to see if we must have a note that Zonemaster assumes it to be turned off.

Reply:
It's optional already, and it has been since at least 0.71 which is dated at 2015-02-18. It does provide a way to test how Zonemaster Engine behaves in the light of inconsistently capitalized domain names in responses.

Reply:
It is disabled by default. Have you tested what it means to turn it on? Zonemaster has a test case where capitalization is tested. Is there any good reason why we should lift this up as an option?

Reply:
I've only verified that it compiles on at least one platform. Nothing more.

The message+description of the commit where it was introduced is the best documentation I've found: dca14a4f4af0013f7325153eb149f02813ef06bb. I'll update the feature description to spell out that it capitalization regards returned domain names.

If you think it should be removed, please file an issue. This PR is about doing as little as possible to make the IDN feature explicit and documented and it has already attracted a fair amount of collateral changes.